src/browser/fake_network.js

```diff
@@ -43,6 +58,220 @@ function iptolong(parts) {
     return parts[0] << 24 | parts[1] << 16 | parts[2] << 8 | parts[3];
 }
 
+class Uint8Stream
+{
+    /**
+     * @param {number} initial_capacity
+     * @param {number} maximum_capacity
+     */
+    constructor(initial_capacity, maximum_capacity)
+    {
+        this.maximum_capacity = maximum_capacity;
```

Comment:
Unless we need ringbuffers with different capacities, I'd suggest making this a global constant.

Reply:
Class [NetworkAdapter](https://github.com/copy/v86/blob/ba7f9405840b6ae476bf01a5b4dfb7f1a6e48055/src/browser/network.js#L21) might be another candidate that could benefit from class GrowableRingbuffer, as well as the inverse TCP data flow direction (from guest to internet) which I haven't much looked at so far.

I have yet to make up my mind what to use for `maximum_capacity`. It is clear that when it gets hit (or if Uint8Array constructor throws a RangeException) the whole TCP connection needs to be torn down, dropping traffic is not an option because there is no resend mechanism at this point.

I'll come back to this.